powersimdata/input/export_data.py

```diff
@@ -140,3 +149,281 @@ def export_transformed_profile(kind, scenario_info, grid, ct, filepath):
     profile = tp.get_profile(kind)
     print(f"Writing scaled {kind} profile to {filepath} on local machine")
     profile.to_csv(filepath)
+
+
+def export_to_pypsa(scenario, preserve_all_columns=False):
+    """Export a Scenario/Grid instance to a PyPSA network.
+
+    .. note::
+        This function does not import storages yet.
+
+    :param scenario powersimdata.input.grid.Grid/
+        powersimdata.scenario.scenrario.Scenario:
+        Input object. If a Grid instance is passed, operational values
+        will be used for the single snapshot "now".
+        If a Scenario instance is passed, all available time-series will be
+        imported.
+    :param preserve_all_columns bool: Whether to import all columns
+        of the corresponding component. If true, this will also
+        import columns that PyPSA does not process. The default
+        is False.
+
+    """
+    from powersimdata.scenario.scenario import Scenario  # avoid circular import
+
+    if not pypsa_available:
+        raise ImportError("PyPSA is not installed.")
+
+    if isinstance(scenario, Grid):
+        grid = scenario
+        scenario = None
+    elif isinstance(scenario, Scenario):
+        grid = scenario.get_grid()
+    else:
+        raise TypeError(
+            "Expected type powersimdata.Grid or powersimdata.Scenario, "
+            f"get {type(scenario)}."
+        )
+
+    drop_cols = []
+
+    # BUS, LOAD & SUBSTATION
+    bus_rename = {
+        "lat": "y",
+        "lon": "x",
+        "baseKV": "v_nom",
+        "type": "control",
+        "Gs": "g_pu",
+        "Bs": "b_pu",
+    }
+
+    bus_rename_t = {
+        "Pd": "p",
+        "Qd": "q",
+        "Vm": "v_mag_pu",
+        "Va": "v_ang",
+    }
+
+    if not preserve_all_columns:
+        drop_cols = ["Vmax", "Vmin", "lam_P", "lam_Q", "mu_Vmax", "mu_Vmin"]
+
+        if scenario:
+            drop_cols += list(bus_rename_t)
+
+    buses = grid.bus.rename(columns=bus_rename)
+    buses.control.replace([1, 2, 3, 4], ["PQ", "PV", "slack", ""], inplace=True)
+    buses["zone_name"] = buses.zone_id.map({v: k for k, v in grid.zone2id.items()})
+    buses["substation"] = "sub" + grid.bus2sub["sub_id"].astype(str)
+    buses["zone_name"] = buses.substation.map({v: k for k, v in grid.zone2id.items()})
+
+    # ensure compatibility with substations (these are imported later)
+    buses["is_substation"] = False
+    buses["interconnect_sub_id"] = -1
+    buses["name"] = ""
```

Comment:
This function is pretty large. What do you think about separating constants from logic to keep the logic part shorter? E.g.
putting into the module main:
```
pypsa_const = {
    "bus": {
        "rename": foo,
        "rename_t": bar,
        "default_drop_cols": baz,
    },
    "generator": {...},
    "branch": {...},
    ...
}
```
and then just accessing these within the function rather than defining them? Alternatively, we could chunk the main function into smaller functions for `bus`, `generator`, `branch` etc. and then use the main function to delegate logic to these.

Reply:
Yes! That is a good idea.